src/manifest.json

```diff
@@ -30,7 +30,8 @@
     "storage",
     "tabs",
     "webRequestBlocking",
-    "webRequest"
+    "webRequest",
+    "bookmarks"
```

Comment:
Let's try to put this under `optional_permissions` and trigger a request for the permission at the right time. @Rob--W based on your experience with users of your add-on, can you think of a good time to prompt for this permission?

Reply:
Usually, an [options page](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Options_pages) would be a good place for it. But this extension doesn't have one. The only UI surface that the add-on has is the browserAction popup, which doesn't look like a good place for this functionality.

Given the required effort (and cost of commitment to maintenance), I wonder whether it might be a better idea to not add the feature for now. Users can just install the other add-on if they want bookmark menu items.

Reply:
Rob, I think I'll still finish it, so that it's an applicable Outreachy application (assuming that's how this works). I guess after that it'll be up to the MAC team whether they want to fully implement it or not.

I was thinking I had two options here:

I can either add a button on the first page of the popup (like opposite the sort tabs button) to enable the bookmark context menu, which I'm not sure is a logical place for this.

Or

I can add an options page, which seems easy enough to do. After that if there's time, I could add a new intro panel to the popup that asks the user if they want to enable the feature (and submits the request).

Right now I have it working while checking if the permissions are enabled, I think I'm nearly done adding an options page, and that matches Rob's suggestion, so I think I'll keep working that way unless @groovecoder has a different suggestion.

Reply:
The feature is not prominent enough to warrant an extra button in the popup (which currently looks relatively minimal & clean).

If you still want to add this feature somewhere, then a new options page would probably be the most fitting place.

Reply:
Newest commit adds the options page.